geoh5py/data/reference_value_map.py

```diff
@@ -40,13 +51,31 @@ def __call__(self):
         return self._map
 
     @property
-    def map(self):
+    def map(self) -> dict[int, str]:
         """
-        :obj:`dict`: A reference dictionary mapping values to strings
+        A reference dictionary mapping values to strings.
+        The keys are positive integers, and the values description.
+        The key '0' is always 'Unknown'.
         """
         return self._map
 
     @map.setter
-    def map(self, value):
-        assert isinstance(value, dict), "Map values must be a dictionary"
+    def map(self, value: dict[int, str]):
+        if not isinstance(value, dict):
+            raise TypeError("Map values must be a dictionary")
+
+        if not all(
+            isinstance(k, (int, np.int16, np.int32, np.int64)) and k >= 0
+            for k in value.keys()
+        ):
+            raise KeyError("Map keys must be positive integers")
+
+        if 0 in value.keys():
+            if value[0] != "Unknown" and not (
+                len(value) <= 2 and value[0] == "False" and value[1] == "True"
+            ):
+                raise ValueError("Map value for 0 must be 'Unknown'")
+        else:
```

Comment:
Could make an explicit loop and call the static method to check each key-value pairs. Only need to check/append the 0:"Unknown" entry beforehand.